### PR TITLE
fix: bump required `dune` and introduce `dune-workspace`

### PR DESCRIPTION
**Pull request by ggreif**
This factors out the `js_of_ocaml` flags into a central location and enables better control
(also fixing a problem when linking under `dune-3.2`).

@crusso has seen the problem on `master` when doing
```
make -C src moc_interpreter.js
```
that compilation flags were passed to the linker of jsoo:
```
js_of_ocaml: unknown option `--no-extern-fs'.
             unknown option `--no-source-map'.
             unknown option `--target-env'.
Usage: js_of_ocaml link [OPTION]... [JS_FILES]...
```

### CODE REVIEW

**Comment by ggreif** [comment]
@rvanasa with some luck this may even reduce the `.js` sizes further, as now more files are compiled with frugal flags.

**Review by crusso — APPROVED** [review]
LGTM - why do the files includes need to change?

**Review by nomeata — APPROVED** [review]
Rubber stamp, looks plausible

File: src/exes/candid_tests.ml [hunk 1]
```diff
@@ -42,14 +42,6 @@ let argspec = Arg.align
 
 module Pretty = Type.MakePretty(Type.ElideStamps)
 
-let load_file f =
```
Comment (ggreif) [inline src/exes/candid_tests.ml:45]: this binding was unused

**Comment by ggreif** [comment]
> LGTM - why do the files includes need to change?

Probably because the filenames are now workspace-relative. I dunno exactly. Maybe before each directory was its own (implicit) workspace.

**Review by rvanasa — APPROVED** [review]
Looks good; thanks for the solution.

**Comment by crusso** [comment]
Just checked it out locally and can verify that make moc_interpreter.js (and make preview in doc) seem to work and create working  js.

**Pull request merged**